lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -912,29 +912,46 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     }
     return hasPrintedAnything;
   }
-    
-  void printRenameForDecl(const AvailableAttr *AvAttr, const ValueDecl *D,
-                          bool includeQuotes) {
-    assert(!AvAttr->Rename.empty());
-    
-    auto renamedParsedDeclName = parseDeclName(AvAttr->Rename);
-    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
-    
+  
+  const ValueDecl *getRenameDecl(const ValueDecl *D, const ParsedDeclName renamedParsedDeclName) {
     auto declContext = D->getDeclContext();
-    const ValueDecl *renamedDecl = nullptr;
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
```

Comment:
Nitpick: `D->getASTContext()` is used multiple times; it would probably be best to save a reference to it early on.